feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -181,6 +181,35 @@ static void radio_state_custom_options_get(struct schema_Wifi_Radio_State *rstat
 	}
 }
 
+static void set_channel_max_power(struct schema_Wifi_Radio_State *rstate,
+                                    int *index, int channel,
+                                    int max_power)
+{
+	rstate->channel_max_power_keys[*index] = channel;
+	rstate->channel_max_power[*index] = max_power;
+	*index += 1;
+	rstate->channel_max_power_len = *index;
+}
+
+// Update the entire channel_max_power map in the radio state
+static void update_channel_max_power(char* phy, struct schema_Wifi_Radio_State *rstate) {
+	int channels[64];
+	int channel_count = phy_get_channels(phy, channels);
+	
+	// Clear the data previously stored in channel_max_power
+	memset(rstate->channel_max_power_keys, 0, sizeof(rstate->channel_max_power_keys));
+	memset(rstate->channel_max_power, 0, sizeof(rstate->channel_max_power));
+	rstate->channel_max_power_len = 0;
+
+	// Set the new values for channel_max_power
+	int i, index = 0;
+	for (i = 0; i < channel_count; i++) {
```

Comment:
This loop runs over the number of channels we have without any test of the size of the destination buffer.
It may have been fine in your testing and break in some condition where lots of channels are available.

Reply:
Added check to `i` so it would max out at 64